main/modes/games/atrium/atrium.c

```diff
@@ -0,0 +1,1292 @@
+    {
+        freeWsg(&atr->uiElements[idx]);
+    }
+    heap_caps_free(atr->uiElements);
+    for (int idx = 0; idx < ARRAY_SIZE(sonaBodies); idx++)
+    {
+        freeWsg(&atr->bodies[idx]);
+    }
+    heap_caps_free(atr->bodies);
+    heap_caps_free(atr);
+}
+
+static void atriumMainLoop(int64_t elapsedUs)
+{
+    buttonEvt_t evt = {0};
+    switch (atr->state)
+    {
+        case ATR_TITLE:
+        {
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down) // if the button is pressed down on the title screen
+                {
+                    if ((evt.button & PB_A))
+                    {
+                        atr->state = ATR_DISPLAY;
+                    }
+                    else if ((evt.button & PB_B))
+                    {
+                        atr->state = ATR_EDIT_PROFILE; // if B is pressed, go to edit profile view
+                    }
+                }
+            }
+            drawAtriumTitle(elapsedUs);
+            atr->loadedProfs = false;
+            shuffleSonas();
+
+            break;
+        }
+        case ATR_DISPLAY:
+        {
+            // Handle input
+            while (checkButtonQueueWrapper(&evt))
+            {
+                atr->lastPage = atr->page;
+
+                if (evt.down)
+                {
+                    if (evt.button & PB_UP)
+                    {
+                        atr->up = true;
+                    }
+                    else if (evt.button & PB_LEFT)
+                    {
+                        atr->left = true;
+                        atr->page--;
+                        if (atr->page < 0)
+                        {
+                            atr->page = 0;
+                        }
+                    }
+                    else if (evt.button & PB_RIGHT)
+                    {
+                        atr->right = true;
+                        atr->page++;
+                        if (atr->page > (atr->numRemoteSwsn - 1) / 4)
+                        {
+                            atr->page = (atr->numRemoteSwsn - 1) / 4;
+                        }
+                    }
+                    else if (evt.button & PB_DOWN)
+                    {
+                        atr->down = true;
+                    }
+                    if (evt.button & PB_UP)
+                    {
+                        atr->up = false;
+                        if (atr->lbState > 0)
+                        {
+                            atr->lbState--;
+                        }
+                        atr->loadAnims = 0;
+                    }
+                    else if (evt.button & PB_DOWN)
+                    {
+                        atr->down = false;
+                        if (atr->lbState < 2)
+                        {
+                            atr->lbState++;
+                        }
+
+                        atr->loadAnims = 0;
+                    }
```

Comment:
This doesn't pass the sniff test. It's an `if`, not an `else if` without a line break between, and `PB_UP` and `PB_DOWN` are checked in both blocks, so `atr->up` and `atr->down` are both set and then immediately overwritten.

What was the intent here? The effect is equivalent to deleting the prior button checks and turning this into one large `if`/`else if` statement.